### PR TITLE
Add model associations and specs

### PR DESCRIPTION
**Pull request by apalski**
PR for model associations
- add associations to user, budget, income and expense
- create models budget_expense and budget_income and their tables
- add user_id to budgets table
- run migrations
- add association specs to user, budget, income and expense model specs
Looking for feedback on all aspects, particularly whether there are other specs I should have.
Thanks for looking at this PR for me!
https://trello.com/c/Z7pbLZZV

### CODE REVIEW

File: spec/models/budget_spec.rb [hunk 1]
```diff
@@ -8,6 +8,14 @@
     it { is_expected.to validate_uniqueness_of(:name) }
   end
 
+  context "associations" do
+    subject { build(:budget) }
```
Comment (elle) [inline spec/models/budget_spec.rb]: No need for the subject here. So you should be able to remove lines 12-13

File: spec/models/expense_spec.rb [hunk 2]
```diff
@@ -8,4 +8,10 @@
     it { is_expected.to validate_uniqueness_of(:name) }
     it { is_expected.to validate_presence_of(:frequency) }
   end
+
+  context "associations" do
+    subject { build(:expense) }
+
```
Comment (elle) [inline spec/models/expense_spec.rb]: Same here... No need for the subject.

File: spec/models/income_spec.rb [hunk 3]
```diff
@@ -7,4 +7,10 @@
     it { is_expected.to validate_uniqueness_of(:name) }
     it { is_expected.to validate_presence_of(:frequency) }
   end
+
+  context "associations" do
+    subject { build(:income) }
+
+    it { is_expected.to have_many(:budgets).through(:budget_incomes) }
```
Comment (elle) [inline spec/models/income_spec.rb]: And here...

File: spec/models/user_spec.rb [hunk 4]
```diff
@@ -11,6 +11,12 @@
     it { is_expected.to have_secure_password }
   end
 
+  context "associations" do
+    subject { build(:user) }
+
+    it { is_expected.to have_many(:budgets) }
```
Comment (elle) [inline spec/models/user_spec.rb]: And here :)

File: spec/models/budget_spec.rb (hunk 1 above)
Comment (apalski) [reply spec/models/budget_spec.rb]: Done in sha: 1453820

File: spec/models/expense_spec.rb (hunk 2 above)
Comment (apalski) [reply spec/models/expense_spec.rb]: Done in sha: 1453820

File: spec/models/income_spec.rb (hunk 3 above)
Comment (apalski) [reply spec/models/income_spec.rb]: Done in sha: 1453820

File: spec/models/user_spec.rb (hunk 4 above)
Comment (apalski) [reply spec/models/user_spec.rb]: Done in sha: 1453820

**Pull request closed without merging**